### DISCUSSION TITLE
linux demo是否存在内存泄漏

### DISCUSSION THREAD

**Discussion by jackTJT** [Tencent/libpag#2122] [General] [answered]
==2858594== HEAP SUMMARY:
==2858594==     in use at exit: 109,087,838 bytes in 7,580 blocks
==2858594==   total heap usage: 655,367 allocs, 647,787 frees, 240,488,102 bytes allocated
==2858594== 
==2858594== 4 bytes in 1 blocks are possibly lost in loss record 15 of 4,201

在使用valgrind运行时发现demo好像有内存泄漏问题
在创建以下变量时，只有pagPlayer需要释放，pagSurface不用额外调用函数去释放吗
auto pagSurface = pag::PAGSurface::MakeOffscreen(pagFile->width(), pagFile->height());
  auto pagPlayer = new pag::PAGPlayer();
  pagPlayer->setSurface(pagSurface);
  pagPlayer->setComposition(pagFile);

**Comment by kevingpqi123** [comment, marked as answer]
感觉你没有找到原因，建议系统了解下 C++ 的内存管理机制， PAGSurface 是通过静态方法创建的，返回的是一个 shared_ptr 的智能指针
